modules/server-node/src/services/store.ts

```diff
@@ -295,6 +295,48 @@ export class PrismaStore implements IServerNodeStore {
     this.prisma = new PrismaClient(_dbUrl ? { datasources: { db: { url: _dbUrl } } } : undefined);
   }
 
+  /// Retrieve all tx's that have been submitted, but were not confirmed/mined
+  /// (and did not fail).
+  async getActiveTransactions(): Promise<StoredTransaction[]> {
+    const activeTransactions = await this.prisma.onchainTransaction.findMany({
+      where: {
+        transactionHash: {
+          not: undefined
+        },
+        // TODO: Any other key fields that we know HAVE to be defined for receipts that we should include here?
+        // If these key receipt fields are undefined, then we know we never got
+        // a valid receipt for this transaction.
+        blockHash: undefined,
+        gasUsed: undefined,
+      }
+    });
```

Comment:
we have a `status` field that is `submitted` when it's submitted, and `mined` or `failed` otherwise. The only race around this status is when it was fired off, but not saved in the db. Maybe we need to save the `MinimalTransaction` under a different status before submitting to be extra careful? 

however, the status should be used as it is more clear.

Reply:
If it's fired off but not saved in the db we don't save it and I don't think we should tbh.